nucypher/network/nodes.py

```diff
@@ -543,7 +542,8 @@ def select_teacher_nodes(self):
         nodes_we_know_about = self.known_nodes.shuffled()
 
         if not nodes_we_know_about:
-            raise self.NotEnoughTeachers("Need some nodes to start learning from.")
+            self.log.warn("Need some nodes to start learning from.")
+            return False
```

Comment:
If we are here, does it mean that we don't know about any nodes at all i.e. we can't learn and are stuck in a state we'd potentially never get out of i.e. without a teacher, we can't learn anything new, and without learning anything new, we can never have a new teacher...? Or am I mistaken.

Reply:
That's correct.  The caller handles an empty collection and raises an error.

Reply:
Is the caller `cycle_teacher_node()`? If so, it no longer raises...

Reply:
Hmm you're right.  At the time I originally wrote this PR back in October, I wanted to permit this node to continue running for some reason that I cannot remember now... it also looks like this may result in some kind of unhanded error, but we don't see that in CI failure.  Also, this function and area has incoming moderate rework/refactor in PR #3382

Reply:
```suggestion
            raise self.NotEnoughTeachers("Need some nodes to start learning from.")
```